### ISSUE TITLE
Consolidating Gradient Controls 

### ISSUE THREAD

**Issue by pablohoneyhoney** [WordPress/gutenberg#21269]
As we evolve the sidebar to [the new UI language](https://github.com/WordPress/gutenberg/issues/18667), here a smaller step to consolidate some UI controls within gradient that seem a bit disconnected, unpolished, and congested. 

**Current**

<img width="288" alt="current-gradient" src="https://user-images.githubusercontent.com/25516611/77937558-828fd280-7282-11ea-881e-5ed6aa6427b0.png">

**New**

- Removing labels as the functions are in context of Color / Gradient. 
- Gradient types becoming its own label and dropdown, to allow scalability to angular, diamond on top of linear and radial. 
- Aligning styles. 

Here some permutations

<img width="290" alt="new-gradient-controls" src="https://user-images.githubusercontent.com/25516611/77938330-938d1380-7283-11ea-9c30-6049d3a62c8f.png">

The one that seems working better and more balanced

<img width="290" alt="new-gradientcontrols" src="https://user-images.githubusercontent.com/25516611/77938587-f54d7d80-7283-11ea-82ea-5ebab3b914ab.png">

**Comment by pablohoneyhoney** [comment]
Here another iteration in case we do need to keep labels. 

<img width="290" alt="new-gradientcontrols" src="https://user-images.githubusercontent.com/25516611/77945554-d30d2d00-728e-11ea-88b6-afef8f7a5999.png">

**Comment by mapk** [comment]
Thanks for including a version with labels, @pablohoneyhoney. 

1. The dropdown you've got is so much cleaner and communicates the gradient type just fine.
2. I really like the "Angle" label aligning to the edge of the angle degree input field. This adds the needed balance.
3. Of the 3 angle dial controls, I prefer the black dot. It ties in nicely with the other fields.
4. My only hesitation here is the capitalization of the labels. All caps can cause readability issues for some. Worth getting more feedback on that.

**Comment by ItsJonQ** [comment]
@pablohoneyhoney This is lovely! I started working on this 😊 

First step is refactoring the `AnglePickerControl` a bit with the new styles. The JS works quite well, so I didn't need to do much there. I've also replaced the regular `input` with our new G2ified `NumberControl`, which supports features like drag-to-update and <kbd>Shift</kbd> key jumping.

![2020-06-24 16-57-01 2020-06-24 16_57_16](https://user-images.githubusercontent.com/2322354/85626978-d587e480-b63b-11ea-83f4-ee80c2803657.gif)

I pushed things here for now:
https://github.com/WordPress/gutenberg/tree/update/gradient-controls

(Still WIP of course)

Next is to update the `<SelectControl />` with G2 styles.

And lastly... putting it all together :D

**Comment by ItsJonQ** [comment]
Updates!


<img width="327" alt="Screen Shot 2020-06-29 at 4 27 07 PM" src="https://user-images.githubusercontent.com/2322354/86052799-8b30aa00-ba25-11ea-820c-6de0a882b62f.png">

Working on a new `<SelectControl />`. Style wise, it looks exactly like our new `InputControl`, which has the new G2 styles ✨ . It also has the same props, like `size="small"`, and other features (e.g. floating labels). Last detail... The down arrow is using the `chevronDown` from `@wordpress/icons`, rather than the default HTML arrow. This gives us more UI control to ensure consistency across platforms.

GIF Demo:

![demo](https://user-images.githubusercontent.com/2322354/86052782-84a23280-ba25-11ea-9161-7b3ed1a07320.gif)

**Comment by jasmussen** [comment]
This is looking really fine. I'll try and work on some consistency with buttons in the sidebar today, which should help us get closer.

**Comment by jasmussen** [comment]
I was needing a gradient yesterday, and wanted one that was really sharp, as in not really a gradient at all, just two solid colors in a diagonal. But the draggable controls can't overlap:

![gradient](https://user-images.githubusercontent.com/1204802/86231798-5f333700-bb93-11ea-85f0-651b35e7cc00.gif)

Can we allow this? It seems like a fair use case. Or even if I wanted to flip the gradient entirely, move the orange color all the way to the right past the red color?

**Comment by pablohoneyhoney** [comment]
Looking good!

Worth looking at proportions (perhaps taller dragging area so it becomes hierarchically more prominent with a bigger area of interaction) and more distance between the slider and the controls below.

**Comment by paaljoachim** [comment]
I am closing this issue as @ItsJonQ Jon has made a PR which has also been merged. 
Please reopen if I am mistaken.
